src/stream.rs

```diff
@@ -25,28 +25,32 @@ impl<'a> WrappedSlice<'a> {
     self.offset
   }
 
-  fn consume(&mut self, n_bytes: usize) -> usize {
+  fn consume(&mut self, n_bytes: usize) -> Option<usize> {
     let offset = self.offset.saturating_add(n_bytes);
     // Make sure `offset <= buf.len()`
-    self.offset = std::cmp::min(self.buf.len(), offset);
-    self.offset
+    if offset <= self.buf.len() {
+      self.offset = offset;
+      Some(offset)
+    } else {
+      None
+    }
```

Comment:
This would stop partial reads from working (i.e. requesting 1024 bytes, but the stream only has 128 bytes).  I think it would be better to return `None` if there was no more bytes available `self.offset == self.buf.len()`.

Reply:
I just tested it with fseek and it indeed truncates.